application/models/Traits/PermissionTrait.php

```diff
@@ -31,6 +31,29 @@ public static function getMinimalPermissionRead()
         return null;
     }
 
+    /**
+     * get criteria from Permission
```

Comment:
This description could be enhanced a bit. When this method is supposed to be used?
Sorry to insist on this documentation, but these kind of methods are not trivial and could be missused / miss interpretated in the future.

Reply:
Currently : used in grid, and list.

When you have a list of Objects to shown : you can not check Permission on each single Object due to performance issue.

Reply:
can be used with findByPk but i think a clear 'read' permission check is _in general_ better to throw appropriate error (403 VS 404)

Reply:
ok, so we should add something like this in the doc ...?
> Returns a criteria which would flter and return rows for which the user has read/list access.
> Used in grid, and list. When you have a list of Objects to show, you can not check Permission on each single Object due to performance issue. This method is used then.